xtask/src/xtask.rs

```diff
@@ -186,14 +186,26 @@ fn test_tmt(sh: &Shell) -> Result<()> {
                 Ok(acc)
             })?;
     all_plan_files.sort_by_key(|v| v.0);
-    println!("Discovered plans: {all_plan_files:?}");
+    Ok(all_plan_files)
+}
+
+#[context("test-integration")]
+fn test_tmt(sh: &Shell) -> Result<()> {
+    let tests = match std::env::var("TMT_TEST") {
+        Ok(test) => {
+            vec![(0, test)]
+        }
+        Err(_) => all_plan_files(sh)?,
+    };
+
+    println!("Discovered plans: {tests:?}");
```

Comment:
Now when `TMT_TEST` is set we'll now only say that we discovered what they asked for right?

How about doing this like:
```
let mut tests = all_plan_files()?;
if let Ok(name) std::env::var("TMT_TEST") {
    tests.retain(|x| x.1.as_str() == name);
    if tests.is_empty() {
        anyhow::bail!("Failed to match test: {name}");
    }
}
```

or so

Reply:
good idea, updated